packages/cardpay-reward-programs/cardpay_reward_programs/rules/min_other_merchants_paid.py

```diff
@@ -39,25 +39,21 @@ def df_to_payment_list(
         new_df = df.copy().groupby("payee").agg({"merchant": "nunique"}).reset_index()
         new_df["rewardProgramID"] = reward_program_id
         new_df["paymentCycle"] = payment_cycle
-        new_df["validFrom"] = self.end_block
```

Comment:
The end block here wouldn't be right - a cashback program that is calculated each month for a year should be claimable before the end of the whole year.

Reply:
I **should** have put `payment_cycle`.

Reply:
Strongly Agreed. I **should** have put `payment_cycle` here